go-controller/pkg/cni/helper_linux.go

```diff
@@ -563,12 +581,12 @@ func (pr *PodRequest) UnconfigureInterface(ifInfo *PodInterfaceInfo) error {
 		klog.Warningf("Unexpected configuration %s, Device ID must be present for pod request on smart-nic host", podDesc)
 		return nil
 	}
-	// 1. For SRIOV case, we'd need to move the netdevice from container namespace back to the host namespace
+	// 1. For SRIOV case, if VF is not VFIO,  we'd need to move the netdevice from container namespace back to the host namespace
 	// 2. If it is secondary network and non-dpu mode, then get the container interface index
 	//    so that we know the host-side interface name.
 	ifnameSuffix := ""
 	isSecondary := pr.netName != types.DefaultNetworkName
-	if pr.CNIConf.DeviceID != "" || (isSecondary && !ifInfo.IsDPUHostMode) {
+	if (pr.CNIConf.DeviceID != "" && !pr.IsVFIO) || (isSecondary && !ifInfo.IsDPUHostMode) {
```

Comment:
@winsopc 

Can you please fix this per our discussion on slack?

Essentially, for the VFIO case there is nothing to be done inside the namespace since that device is not present.